backends/cadence/aot/quantizer/patterns.py

```diff
@@ -44,18 +47,20 @@ class PartitionAnchors:
 
 class QuantizationPattern(ABC):
     @abstractmethod
-    def partition_types(self):
+    def partition_types(self) -> list[OpOverload]:
```

Comment:
I think we need to support Python 3.8 here, which doesn't support `list[x]`, and you need to use `typing.List[x]`